### DISCUSSION TITLE
Deprecate the use of words Whitelist and Blacklist

### DISCUSSION THREAD

**Discussion by dschveninger** [golangci/golangci-lint#2869] [General] [answered]
I was wonder if you would be opposed to deprecating the use of the words blacklist and whitelist in this repository?

I would like to get feed back on either addressing the whole repo

Whitelist https://github.com/golangci/golangci-lint/search?q=whitelist
Blacklist https://github.com/golangci/golangci-lint/search?q=blacklist

Or just address it with the config words. The approach is to mirror the old config with a new config. The old config would echo out a deprecation message for a future version.

if you are not opposed to the idea I may have someone address this. I just wanted to get your feedback.

**Comment by ldez** [comment, marked as answer]
Hello,

Golangci-lint is a wrapper around linters. These words come from linters themself. Some of them that we already depreciate these options.

We follow the linter configuration due to the complexity of manipulation which can be significant in some cases, and we don't want to increase the complexity of maintenance (our time is limited).

It's not possible to deprecate all the uses without breaking things, so the changes can be done but they must be done very carefully.

So, we are not opposed to this kind of depreciation, but I want to be clear: it's not possible to remove all the uses of these words.

**Reply by dschveninger** [reply]
Thank you for your response. We will talk to the individual linters  and see if we can get those projects to consider the request.
